homeassistant/components/twitch/coordinator.py

```diff
@@ -83,11 +83,23 @@ async def _async_update_data(self) -> dict[str, TwitchUpdate]:
             False,
         )
         data = {}
+        streams: list[Stream] = [
+            s
+            async for s in self.twitch.get_followed_streams(
+                user_id=self.current_user.id, first=100
+            )
+        ]
+        follows: list[FollowedChannel] = [
+            f
+            async for f in await self.twitch.get_followed_channels(
+                user_id=self.current_user.id, first=100
+            )
+        ]
```

Comment:
can we make those a dict comprehension so we don't have to loop over almost the full list for every channel?

Reply:
I'm sure I could figure out how (if you have tips, it'd be appreciated), but I don't think I'll be able to get to it until after this weekend.

Reply:
That's fine.

```
streams = {s.stream_id: s async for s in ...}
```
to give some pseudo code. This way we can just do `streams[id]`